### PR TITLE
Remove non-active facets from facet display on channel page.

### PR DESCRIPTION
**Pull request by ekes**

### CODE REVIEW

**Comment by finnlewis** [comment]
Need to take this for a test drive.

**Comment by ekes** [comment]
pre-MM ping

**Comment by ekes** [comment]
@finnlewis post-MM ping

**Comment by finnlewis** [comment]
@ekes will take a look at the failing test. 

If anyone else can do a manual test, that would be great ! @Adnan-cds @stephen-cox @andybroomfield

File: src/DirectoryExtraFieldDisplay.php [hunk 1]
```diff
@@ -286,6 +298,19 @@ public function preprocessFacetList(array &$variables) {
...
+
+    // This is usually on a channel node. If so remove facets not active on
+    // channel.
+    if (($channel = $this->routeMatch->getParameter('node'))
+      && $channel instanceof NodeInterface
+      && $channel->bundle() == 'localgov_directory'
+    ) {
+      $active_facets = array_column($channel->localgov_directory_facets_enable->getValue(), 'target_id');
+    }
+    if ($active_facets) {
```
Comment (Adnan-cds) [inline src/DirectoryExtraFieldDisplay.php]: Can this lead to a PHP warning if we are not on a localgov_directory node?

**Comment by Adnan-cds** [comment]
> If anyone else can do a manual test, that would be great !

Tried this on a test site with a small dataset and it worked as expected :)

File: src/DirectoryExtraFieldDisplay.php (hunk 1 above)
Comment (ekes) [reply src/DirectoryExtraFieldDisplay.php]: Yep. That's one of the test fails. Should be updated with [31db4e5](https://github.com/localgovdrupal/localgov_directories/pull/270/commits/31db4e5a6f16dbf04f653926c29766ae8b0459e5)

File: src/DirectoryExtraFieldDisplay.php [hunk 2]
```diff
@@ -291,6 +303,20 @@ public function preprocessFacetList(array &$variables) {
...
+    // This is usually on a channel node. If so remove facets not active on
+    // channel.
+    $active_facets = NULL;
+    if (($channel = $this->routeMatch->getParameter('node'))
+      && $channel instanceof NodeInterface
+      && $channel->bundle() == 'localgov_directory'
+    ) {
+      $active_facets = array_column($channel->localgov_directory_facets_enable->getValue(), 'target_id');
+    }
+    if (!is_null($active_facets)) {
```
Comment (Adnan-cds) [inline src/DirectoryExtraFieldDisplay.php:316]: I am okay with this.  Dropping line 309 and changing 316 to `if (!empty($active_facets)) {` would also do.  Sorry, should have mentioned this earlier :(

**Review by Adnan-cds — APPROVED** [review]
Works for me.  Thanks.

File: src/DirectoryExtraFieldDisplay.php (hunk 2 above)
Comment (ekes) [reply src/DirectoryExtraFieldDisplay.php:316]: After reading a blog post, I can't remember where, I'm quite liking avoiding using `empty` and trying to be more explicit. Which in this case means I guess we understand array_column could technically return an empty array and this get passed into array_intersect_key.

**Pull request merged**